tspng/extraction.py

```diff
@@ -110,3 +112,34 @@ def extract_from_files(paths: List[str], mime_type: str=MIME_TYPE) -> Dict:
     for path in paths:
         nested_dict[path] = extract_from_file(path, mime_type)
     return nested_dict
+
+def extract_from_folder(path: str, mime_type: str=MIME_TYPE) -> Dict:
+    '''
+    Returns a nested dictionary of metadata from a folder of TSPNG file paths.
+
+        Parameters:
+                path (list[str]): List of file paths
+                mime_type (str): Optional; Media type of file,
+                    default is 'application/vnd.theiascope.io+json'
+
+        Returns:
+                (dict): Dictionary containing metadata of each file
+
+        Raises:
+                Exception: If path is not a directory
+                Exception: If path does not contain a PNG file
+    '''
+    #check if directory
+    if not os.path.isdir(path):
+        raise Exception(f"{path} is not to a directory.")
+    # return all files as a list
+    file_list = []
+    for file in os.listdir(path):
+        # check the files which are end with specific extension
+        if file.endswith(".png"):
```

Comment:
There are some weird edge cases out there when working with files, extensions, and file names. I would use the [`os.path.splitext`](https://docs.python.org/3/library/os.path.html#os.path.splitext) function to get the file extension and check if it is a `.png` to automatically handle any of those weird edge cases.

Reply:
Change made as requested